src/main/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+ * Authentication is handled using the {@link PasswordEncoder} and the {@link AysTokenService} is used for
+ * generating and refreshing access tokens.
+ */
+@Service
+@RequiredArgsConstructor
+class AdminUserAuthServiceImpl implements AdminUserAuthService {
+
+    private final AdminUserRepository adminUserRepository;
+    private final AdminUserRegisterVerificationRepository adminUserRegisterVerificationRepository;
+    private final AdminUserRegisterRequestToAdminUserEntityMapper adminUserRegisterRequestToAdminUserEntityMapper = AdminUserRegisterRequestToAdminUserEntityMapper.initialize();
+
+    private final OrganizationRepository organizationRepository;
+
+    private final AysTokenService tokenService;
+    private final PasswordEncoder passwordEncoder;
+
+    /**
+     * Registers a new admin user based on the given {@link AdminUserRegisterRequest} object. First, it retrieves
+     * the {@link AdminUserRegisterVerificationEntity} associated with the verificationId provided in the request.
+     * Then, it checks if the requested organization exists, if the email and username are unique, and finally
+     * maps the request to an {@link AdminUserEntity} and saves it to the database. If successful, the verification
+     * entity is completed and saved to the database.
+     *
+     * @param registerRequest the request object containing the information for the new admin user
+     * @throws AysAdminUserRegisterVerificationCodeNotValidException if the verificationId provided is not valid
+     * @throws AysOrganizationNotExistException                      if the requested organization does not exist
+     * @throws AysAdminUserAlreadyExistsByEmailException             if an admin user with the same email already exists
+     * @throws AysAdminUserAlreadyExistsByUsernameException          if an admin user with the same username already exists
+     */
+    @Override
+    public void register(final AdminUserRegisterRequest registerRequest) {
+
+        final AdminUserRegisterVerificationEntity verificationEntity = adminUserRegisterVerificationRepository
+                .findById(registerRequest.getVerificationId())
+                .orElseThrow(() -> new AysAdminUserRegisterVerificationCodeNotValidException(registerRequest.getVerificationId()));
+
+        if (!organizationRepository.existsById(registerRequest.getOrganizationId())) {
+            throw new AysOrganizationNotExistException(registerRequest.getOrganizationId());
+        }
+
+        if (adminUserRepository.findByEmail(registerRequest.getEmail()).isPresent()) {
+            throw new AysAdminUserAlreadyExistsByEmailException(registerRequest.getEmail());
+        }
+
+        if (adminUserRepository.findByUsername(registerRequest.getUsername()).isPresent()) {
+            throw new AysAdminUserAlreadyExistsByUsernameException(registerRequest.getUsername());
+        }
+
+//        TODO : check phone number!
```

Comment:
please fix the todo if possible, and remove this line.

Reply:
Fixed ✅